src/ai/backend/agent/agent.py

```diff
@@ -1198,7 +1201,7 @@ def _rm(path: Path) -> None:
                             KernelId(UUID(body["ID"])),
                             ContainerId(body["CID"]),
                             LifecycleEvent.DESTROY,
-                            body.get("reason", "anomaly-detected"),
+                            KernelLifecycleEventReason.get_or_anomaly_detected(body.get("reason")),
```

Comment:
`get_or_anomaly_detected()` is too specific for this use case only. I think it's better to maintain original structure.
```suggestion
                            body.get("reason", KernelLifecycleEventReason.ANOMALY_DETECTED),
```

Reply:
@kyujin-cho Since I changed the parameter types of some related classes from str to just KernelLifecycleReasonEvent, this is where the problem occurs during type checking. I think that some class's type hint would be changed or additional handling would be needed.  what could be better?

Reply:
You can wrap the value `body.get("reason")` with enum class. How about this? At least this will help you keep type consistency.

```python
class KernelLifecycleEventReason(str, enum.Enum):  # enum.StrEnum is coming in version 3.11.
  @classmethod
  def from_value(cls, value: str) -> Optional["KernelLifecycleEventReason"]:
    try:
      return cls(value)
    except ValueError:
      pass
    return None

...

KernelLifecycleEventReason.from_value(body.get("reason")) or KernelLifecycleEventReason.ANOMALY_DETECTED
```